lightning/src/onion_message/messenger.rs

```diff
@@ -413,6 +413,11 @@ impl Responder {
 		Responder { reply_path }
 	}
 
+	/// Returns the reply path for this [`Responder`].
+	pub fn reply_path(&self) -> BlindedMessagePath {
+		self.reply_path.clone()
```

Comment:
Not sure I understand why this needs to be cloned. Can't we return a reference?

Reply:
Sure, I thought it would be _safer_ not returning a reference

Reply:
oh I see, thank you!
I'd need to do a bigger refactor than I thought. We aren't using events of connection needed or awaiting connection in any of our flows. We currently use `next_onion_message_for_peer` to buffer the message and then dequeue it sending it through lnd. But all the time we do separately the process of trying to connect to peer and then sending messages.

Said so, I'm closing this as is probably our implementation issue.